external-resolver/src/main/scala/org/genivi/sota/resolver/vehicle/VehicleDAO.scala

```diff
@@ -29,4 +29,10 @@ object VehicleDAO {
   def list: DBIO[Seq[Vehicle]] =
     vehicles.result
 
+  def exists(vin: Vehicle.Vin): DBIO[Option[Vehicle]] =
```

Comment:
In classical logic a proposition is either true or false, i.e. boolean. In constructive logic, e.g. type theory, you need to construct the object (the term) that witnesses the truth of the proposition (the type).

It's a fascinating topic: https://en.wikipedia.org/wiki/Constructivism_(mathematics)